libshaderc/src/shaderc_test.cc

```diff
@@ -355,29 +396,20 @@ TEST_F(CompileStringWithOptionsTest, ValuelessMacroCompileOptionsNullPointer) {
 
 TEST_F(CompileStringWithOptionsTest, DisassemblyOption) {
   ASSERT_NE(nullptr, compiler_.get_compiler_handle());
-  shaderc_compile_options_set_disassembly_mode(options_.get());
-  // This should work with both the glslang native disassembly format and the
+  // This should work with both the glslang native assembly format and the
   // SPIR-V tools assembly format.
-  const std::string disassembly_text = CompilationOutput(
-      kMinimalShader, shaderc_glsl_vertex_shader, options_.get());
+  const std::string disassembly_text =
+      CompilationOutput(kMinimalShader, shaderc_glsl_vertex_shader,
+                        options_.get(), OutputType::SpirvAssemblyText);
   EXPECT_THAT(disassembly_text, HasSubstr("Capability Shader"));
   EXPECT_THAT(disassembly_text, HasSubstr("MemoryModel"));
-
-  // The mode should be carried into any clone of the original option object.
-  compile_options_ptr cloned_options(
-      shaderc_compile_options_clone(options_.get()));
-  const std::string disassembly_text_cloned_options = CompilationOutput(
-      kMinimalShader, shaderc_glsl_vertex_shader, cloned_options.get());
-  EXPECT_THAT(disassembly_text_cloned_options, HasSubstr("Capability Shader"));
-  EXPECT_THAT(disassembly_text_cloned_options, HasSubstr("MemoryModel"));
```

Comment:
The old test checked two things: 
- that setting the disassembly option on the options object cause the compiler output to be SPIR-V assembly text.
- that such setting was carried over when that options object was cloned.

However, in the new design disassembling is no longer an attribute on the options object. So it doesn't make sense to check the behaviour on a cloned object.  Cloning the options is orthogonal to calling the shaderc_compile_into_spv_assembly, way up in the Compilation helper class.